CougSat1-ADCS/src/interfaces/CDH.cpp

```diff
@@ -1,31 +1,66 @@
 #include "CDH.h"
 
-#include <CISConsole.h>
-
 /**
  * @brief Construct a new CDH::CDH object
  *
  * @param i2c connected to the C&DH
  */
-CDH::CDH(I2CSlave & i2c) : i2c(i2c) {}
+CDH::CDH(uint8_t addr, PinName sda, PinName scl) : i2c(sda, scl) 
+{
+  i2c.address(addr);
+}
 
 /**
  * @brief Checks if a message is available to read
  *
  * @return true is the PMIC is addressed
  */
-bool CDH::hasMessage() {
+bool CDH::messageReceived() {
   int state = i2c.receive();
-  return state == I2CSlave::ReadAddressed || state == I2CSlave::WriteAddressed;
+  return state == mbed::I2CSlave::WriteAddressed ;
 }
 
 /**
- * @brief Process the message coming from the C&DH
+ * @brief Checks if a message is requested by master
  *
- * @return mbed_error_code_t error code
+ * @return true is the PMIC is addressed
  */
-mbed_error_code_t CDH::processMessage() {
-  // Get message
-  // Switch on commandID and fire off the proper function
-  return MBED_ERROR_UNSUPPORTED;
+bool CDH::messageRequested() {
+  int state = i2c.receive();
+  return state == mbed::I2CSlave::ReadAddressed ;
 }
+
+/**
+ * @brief Reads message from I2CSlave buffer into var message
+ * 
+ */
+void CDH::readCDH()
+{
+  printf("Reading i2c buffer\n Messsage = ");
+  i2c.read(message, MESSAGELENGTH);
+  for (uint8_t i = 0; message[i] != NULL; i++)
```

Comment:
What if the CDH send a zero that is a valid data. You should read the first byte then decide how many more to read based on which command it is coming in